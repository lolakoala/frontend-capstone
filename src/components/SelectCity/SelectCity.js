import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import DropDownMenu from 'material-ui/DropDownMenu';
import MenuItem from 'material-ui/MenuItem';

class SelectCity extends Component {
  constructor(props) {
    super(props);
    this.state = {value: ''};
  }

  handleChange = (event, index, value) => this.setState({value});

  addCity = (event, index, value) => {
    const { currentUser, addCity } = this.props;

    this.handleChange(event, index, value);
    addCity(value, currentUser);
  }

  render() {
    //if currentUser.city redirect to edit profile
    if (this.props.currentUser.city) {
      return <Redirect to="/createProfile" />;
    }

    return (
      <div>
        <h2>Please select your city.</h2>
        <DropDownMenu value={this.state.value} onChange={this.addCity}>
          <MenuItem value="Denver" primaryText="denver" />
        </DropDownMenu>
      </div>
    );
  }
}

export default SelectCity;

SelectCity.propTypes = {
  currentUser: PropTypes.object,
  addCity: PropTypes.func
};
