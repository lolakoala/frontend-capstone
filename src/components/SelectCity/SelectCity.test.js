import React from "react";
import SelectCity from './SelectCity.js';
import { shallow, configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SelectCity', () => {
  const mockFn = jest.fn();
  const wrapper = shallow(<SelectCity
    currentUser={{ email: 'lola@lola.com' }}
    addCity={mockFn}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have instructions', () => {
    const instruction = wrapper.find('h2');

    expect(instruction.length).toEqual(1);
  });

  it('should have drop down menu', () => {
    const ddm = wrapper.find('DropDownMenu');

    expect(ddm.length).toEqual(1);
  });

  // state returns undefined
  it.skip('should change state on change of drop down', () => {
    const ddm = wrapper.find('DropDownMenu');

    ddm.simulate('change', { target: { value: 'Denver' } });
    expect(wrapper.state().value).toEqual('Denver');
  });

  it.skip('should run function on change of drop down', () => {
    const ddm = wrapper.find('DropDownMenu');

    ddm.simulate('change', { target: { value: 'Denver' } });
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
