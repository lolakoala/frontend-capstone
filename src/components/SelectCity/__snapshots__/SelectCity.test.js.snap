// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectCity should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SelectCity
    addCity={[Function]}
    currentUser={
        Object {
            "email": "lola@lola.com",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Please select your city.
</h2>,
        <DropDownMenu
          anchorOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
          animated={true}
          autoWidth={true}
          disabled={false}
          iconButton={<NavigationArrowDropDown />}
          maxHeight={500}
          multiple={false}
          onChange={[Function]}
          openImmediately={false}
          value="Denver"
>
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="Denver"
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="Denver"
          />
</DropDownMenu>,
      ],
      "className": "select-city",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select your city.",
        },
        "ref": null,
        "rendered": "Please select your city.",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorOrigin": Object {
            "horizontal": "left",
            "vertical": "top",
          },
          "animated": true,
          "autoWidth": true,
          "children": <MenuItem
            anchorOrigin={
                        Object {
                                    "horizontal": "right",
                                    "vertical": "top",
                                  }
            }
            checked={false}
            desktop={false}
            disabled={false}
            focusState="none"
            insetChildren={false}
            primaryText="Denver"
            targetOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
            value="Denver"
/>,
          "disabled": false,
          "iconButton": <NavigationArrowDropDown />,
          "maxHeight": 500,
          "multiple": false,
          "onChange": [Function],
          "openImmediately": false,
          "value": "Denver",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "right",
              "vertical": "top",
            },
            "checked": false,
            "desktop": false,
            "disabled": false,
            "focusState": "none",
            "insetChildren": false,
            "primaryText": "Denver",
            "targetOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
            "value": "Denver",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Please select your city.
</h2>,
          <DropDownMenu
            anchorOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
            animated={true}
            autoWidth={true}
            disabled={false}
            iconButton={<NavigationArrowDropDown />}
            maxHeight={500}
            multiple={false}
            onChange={[Function]}
            openImmediately={false}
            value="Denver"
>
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="Denver"
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="Denver"
            />
</DropDownMenu>,
        ],
        "className": "select-city",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Please select your city.",
          },
          "ref": null,
          "rendered": "Please select your city.",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
            "animated": true,
            "autoWidth": true,
            "children": <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="Denver"
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="Denver"
/>,
            "disabled": false,
            "iconButton": <NavigationArrowDropDown />,
            "maxHeight": 500,
            "multiple": false,
            "onChange": [Function],
            "openImmediately": false,
            "value": "Denver",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "Denver",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "Denver",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
